pydra/engine/state.py

```diff
@@ -10,23 +10,106 @@
 
 
 class State:
-    def __init__(self, name, incoming_states=None):
+    # dj: not sure if we will use input_names
+    def __init__(self, name, inputs=None, input_names=None, splitter=None, others=None):
         self.name = name
-        self.ndim = None
-        self._states = None
-        self.states = lru_cache(maxsize=1, typed=True)(self.states)
-        self.incoming_states_ = incoming_states
-
-    def states(self, splitter, inputs):
-        self.ndim = len([1 for val in aux.splitter2rpn(splitter) if val == '*'])
-        self._states = list(aux.splits(splitter, inputs))
-        return self._states
-
-    def named_states(self, splitter, inputs):
-        self.ndim = len([1 for val in aux.splitter2rpn(splitter) if val == '*'])
-        return [{"{}.{}".format(self.name, key):value
-                 for key, value in item.items()}
-                for item in self.states(splitter, inputs)]
+        #self.ndim = None
+        # dj: moved inputs and splitter to init
+        # TODO: will change
+        if inputs:
```

Comment:
is there a reason to store inputs in state? it seems that this will then need to be synced whenever an input is changed at the task level. my preference i think is for inputs to be passed whenever splits is called and outputs of splits is stored in the splitter if necessary.

Reply:
I thought that I had a reason, but can't recall now... so maybe it is outdated